draft-ietf-tls-extended-key-update.md

```diff
@@ -512,13 +512,101 @@ Auth | {CertificateVerify}
+request, it sets the status to `accepted` and includes its own key share.
+If the responder declines the request, it sets the status accordingly and
+does not include the key share. While an extended key update is in progress,
+the responder MUST NOT initiate further key updates.
+
+3. On receipt of the ExtendedKeyUpdateResponse message with `accepted` status,
+the initiator is able to derive a secret key based on the exchanged key shares.
+The NewKeyUpdate message is intentionally an empty structure that triggers
+the transition to new keying material.
+
+4. The initiator transmits the NewKeyUpdate message.
+
+5. Upon receiving a NewKeyUpdate message, the responder MUST update its receive
+keys. It then acknowledges the received message by sending its own NewKeyUpdate
+message. This response both acknowledges the incoming key update and initiates
+a new key update in a single message, as the acknowledgment is piggybacked.
+
+6. After the initiator receives the ACK from the responder, the initiator
+MUST update its send key. With the receipt of the NewKeyUpdate message the
+initiator MUST update its receive keys. The initiator itself MUST acknowledge
+the received NewKeyUpdate message with an ACK message.
+
+7. On receipt of the ACK message the responder updates its send key.
+
+Note that the procedure above aligns with the key update procedure defined in
+DTLS 1.3.
+
+{{dtls-key-update}} shows an example exchange illustrating that successful
+ACK processing updates the keys of the NewKeyUpdate message sender, which is
+reflected in the change of epoch values.
+
+~~~
+Client                                             Server
+
+      /-------------------------------------------\
+     |                                             |
+     |             Initial Handshake               |
+      \-------------------------------------------/
+
+ [Application Data]         -------->
+ (epoch=3)
+
+                            <--------      [Application Data]
+                                                    (epoch=3)
+
+      /-------------------------------------------\
+     |                                             |
+     |              Some time later ...            |
+      \-------------------------------------------/
+
+ [ExtendedKeyUpdateRequest] -------->
+ (epoch=3)
+
+                        <-------- [ExtendedKeyUpdateResponse]
+                                                    (epoch=3)
+
+ [NewKeyUpdate] -------->
+ (epoch 3)
+
+
+                            <--------    [NewKeyUpdate]+[ACK]
+                                                    (epoch=3)
+
+ [ACK]                      -------->
+ (epoch=4)
+
+                            <--------      [Application Data]
+                                                    (epoch=4)
+
+ [Application Data]
+ (epoch=4)                  -------->
+~~~
+{: #dtls-key-update title="Example DTLS 1.3 Extended Key Update Exchange."}
+
```

Comment:
I suggest to show in the diagram when send/recv keys are updated.

Reply:
Will do in a separate PR